### PR TITLE
Adding Nektar++ driftwave 2D Hasegawa-Wakatani example model

### PR DESCRIPTION
**Pull request by matt-graham**
Adds a new example model using [Nektar++ 'driftwave' solver](https://github.com/ExCALIBUR-NEPTUNE/nektar-driftwave) for two-dimensional Hasegawa-Wakatani equations.

The deterministic components dynamics solve two-dimensional Hasegawa-Wakatani equations with spatially correlated additive state noise simulated by solving a Helmholtz equation driven by a Gaussian white noise process. System is simulated on a rectangular spatial domain with periodic boundary conditions, and a regular quadrilaterial mesh using Nektar++ spectral
element method implementation.

Currently for simplicity even though the Nektar++ solvers can be compiled with MPI support, this assumes we just run the solver instances for each particle in a single rank. Could be worth investigating using MPI_Comm_spawn or another approach to allow using MPI at both model and particle filter level.

This implementation only currently supports bootstrap filter as the approach we're using for generating state noise doesn't allow easy evaluation of the quantities needed to implement the additional methods necessary for using the locally optimal proposal.

As we generate the white noise fields used to generate the state noise within Nektar++ and there doesn't seem to be a way to control the random seed used for the (Boost) random number generator used for this, the current implementation won't give consistent outputs even if we fix the seed in the filter parameters. We could potentially try to generate the white noise fields using the Julia random number generator, but this would require some knowledge of how the state degrees of freedom are stored internally by Nektar++ and of the spatial discretisation.

### CODE REVIEW

File: NektarDriftwave/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+The example requires the following to be installed:
+
+- Nektar++ with HDF5 support (tested with [version 5.4](https://gitlab.nektar.info/nektar/nektar/-/tree/v5.4.0)) - [installation instructions](https://www.nektar.info/getting-started/installation/).
+- `nektar-driftwave` solver (tested with [commit `5a6e941`](https://github.com/ExCALIBUR-NEPTUNE/nektar-driftwave/tree/5a6e94131926116590a988058b651e653ac423b7)) - [installation instructions](https://github.com/ExCALIBUR-NEPTUNE/nektar-driftwave?tab=readme-ov-file#compilation--installation).
+- Julia (tested with version 1.9) - [installation instructions](https://julialang.org/downloads/).
+
+## Installation
+
+To set up a Julia environment with all the direct Julia dependencies for the example installed, from the `NektarDriftwave` directory in a local clone of this repository run
```
Comment (matt-graham) [inline NektarDriftwave/README.md:20]: Refactor as a package to allow installing using package manager by passing relevant GitHub URL using form
```
Pkg.add(url="https://github.com/Company/MonoRepo", subdir="juliapkgs/Package.jl)")
```

File: NektarDriftwave/README.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+```Julia
+using ParticleDA
+using HDF5
+using MPI
+
+include("src/nektar_driftwave.jl")
```
Comment (matt-graham) [inline NektarDriftwave/README.md:117]: If refactored as a package, change to `using NektarDriftwave` - would also remove need to be in correct directory when running this and other examples.

File: NektarDriftwave/Project.toml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+Distributions = "31c24e10-a181-5473-b8eb-7969acd0382f"
+Gmsh = "705231aa-382f-11e9-3f0c-b7cb4346fdeb"
+HDF5 = "f67ccb44-e63f-5c2f-98bd-6dc0ccc4ba2f"
+MPI = "da04e1cc-30fd-572f-bb4f-1f8673147195"
+PDMats = "90014a1f-27ba-587c-ab20-58faa44d9150"
+ParticleDA = "61cd1fb4-f4c4-4bc8-80c6-ea5639a6ca2e"
+Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+Tables = "bd369af6-aec1-5ad0-b16a-f7cc5008161c"
+TimerOutputs = "a759f4b9-e2f1-59dc-863e-4aeb61b1ea8f"
+XML = "72c71f33-b9b6-44de-8c94-c961784809e2"
```
Comment (matt-graham) [inline NektarDriftwave/Project.toml:12]: Add compatibility constraints

File: NektarDriftwave/README.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+Building Nektar++ with HDF5 support but without MPI, requires manually commenting out / removing [the lines in the `cmake/ThirdPartyHDF5.cmake` file in the Nektar++ source tree which raise an error if trying to build with HDF5 but no MPI support](https://gitlab.nektar.info/nektar/nektar/-/blob/master/cmake/ThirdPartyHDF5.cmake?ref_type=heads#L13-16). After changing these lines in a local clone of the Nektar++ repository, we successively built Nektar++ with HDF5 but no MPI support on ARCHER2 (using the pre-built Cray HDF5 library on ARCHER2) by running the following from the root of the repository
```
Comment (matt-graham) [inline NektarDriftwave/README.md:40]: Change GitLab link to permalink

File: NektarDriftwave/Project.toml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+[compat]
+CSV = "0.10"
+Distributions = "0.25"
+Gmsh = "0.1"
+HDF5 = "0.16, 0.17"
+MPI = "0.20"
+PDMats = "0.11"
+ParticleDA = "1.1"
+Tables = "1.11"
+XML = "0.3"
```
Comment (matt-graham) [inline NektarDriftwave/Project.toml:27]: @giordano does this looks reasonable in terms of specifying `compat` constraints? I've pinned to minor versions of packages which I was using in locally and have found to work, other than HDF5 which based on previous discussion I think v0.17 should also be safe (I was using v0.16 locally).

Comment (giordano) [reply NektarDriftwave/Project.toml:27]: Yes, that's pretty much what I do as well.

**Pull request merged**